lib/array_equals.rb

```diff
@@ -1,5 +1,25 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  if array1.nil? && array2.nil?
+    true
+  elsif array1.nil? && array2 != nil
+    false
+  elsif array1 != nil && array2.nil?
+    false
+  elsif array1 == [] && array2 == []
+    true
+  elsif array1 == [] && array2 != []
+    false
+  elsif array1 != [] && array2 == []
+    false
+  elsif array1.length == array2.length
+    if array1 == array2
```

Comment:
You also need to traverse the arrays index-by-index and return false if they don't match.  That's the point of this assignment.